internal/proto/local/plugin/host_catalog.proto

```diff
@@ -0,0 +1,30 @@
+syntax = "proto3";
+
+package plugin.sdk.proto;
+
+option go_package = "github.com/hashicorp/boundary/internal/plugin/sdk/proto;proto";
+
+import "google/protobuf/struct.proto";
+
+// HostCatalog represents a host catalog as passed in to a plugin call. 
+message HostCatalog {
+	// The ID of the host catalog.
+	string id = 10;
+
+	// Attributes specific to the host catalog.
+	google.protobuf.Struct attributes = 20;
+	
+  // A set of secrets specific to the host catalog.
+	google.protobuf.Struct secrets = 30;
+}
+
+// HostCatalogPersisted represents state persisted between host
+// catalog calls. Its primary purpose is to store authentication data
```

Comment:
```suggestion
// catalog calls. Its intended purpose is to store authentication data
```

Although the plugin could use this for anything really.

Consider also adding a TODO to specify a size limit for this data.  This is something we should add prior to exposing the plugin SDK publicly since it's easier to ease size restrictions than it is to add them after the fact.

Reply:
Added the TODO!